client/starwhale/api/_impl/data_store.py

```diff
@@ -1880,24 +1880,12 @@ def list_tables(
         self,
         prefixes: List[str],
     ) -> List[str]:
-        table_names = set()
-
-        for prefix in prefixes:
-            prefix_path = Path(self.root_path) / prefix.strip("/")
-            for fpath in prefix_path.rglob(f"*{datastore_table_file_ext}*"):
-                if not fpath.is_file():
-                    continue
-
-                table_name = str(fpath.relative_to(self.root_path)).split(
-                    datastore_table_file_ext
-                )[0]
-                table_names.add(table_name)
-
-            for table in self.tables:
-                if table.startswith(prefix):
-                    table_names.add(table)
-
-        return list(table_names)
+        manifest = self._load_manifest()
```

Comment:
add 'self.lock' for the `list_tables` function ?

Reply:
`load_manifest` will see the single version of the content because of the `move` operation of the file,
So I think the lock is useless here.